`fpm build --flag "-fPIC -ldl"` doesn't work on CI

Error message

´+ gfortran  -fPIC -ldl -J build/gfortran_EC16EF33F7CCE541/fpm-search -I build/gfortran_EC16EF33F7CCE541/fpm-search  build/gfortran_EC16EF33F7CCE541/fpm-search/app_main.f90.o build/gfortran_EC16EF33F7CCE541/fpm-search/app_package_types.f90.o build/gfortran_EC16EF33F7CCE541/fpm-search/app_download_helper.f90.o build/gfortran_EC

/usr/bin/ld: build/gfortran_EC16EF33F7CCE541/fpm-search/libfpm-search.a(build_dependencies_dynload-base_src_dynloadc.c.o): in function `dynload_init':
dynloadc.c:(.text+0x20): undefined reference to dlopen'

On CI `windows-latest` , the error message is

'build\dependencies\M_escape\src\M_escape_initialize.c:17:13: error: 'ENABLE_VIRTUAL_TERMINAL_PROCESSING' undeclared (first use in this function)
     mode |= ENABLE_VIRTUAL_TERMINAL_PROCESSING;
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
build\dependencies\M_escape\src\M_escape_initialize.c:17:13: note: each undeclared identifier is reported only once for each function it appears in'

> On CI `windows-latest` , the error message is
> 
> 'build\dependencies\M_escape\src\M_escape_initialize.c:17:13: error: 'ENABLE_VIRTUAL_TERMINAL_PROCESSING' undeclared (first use in this function)
> mode |= ENABLE_VIRTUAL_TERMINAL_PROCESSING;
> ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> build\dependencies\M_escape\src\M_escape_initialize.c:17:13: note: each undeclared identifier is reported only once for each function it appears in'

Fixed in https://github.com/brocolis/fpm-search/commit/cf06f396a904099be7e05ed491ed4a8cc7f6b21a

--- a/.github/workflows/fpm.yml
+++ b/.github/workflows/fpm.yml
@@ -4,20 +4,43 @@ on: [push, pull_request]
 
 jobs:
   gfortran:
-    runs-on: windows-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        gcc_v: [9] # Version of GFortran we want to use.
+        include:
+        - os: ubuntu-latest
+          os-arch: linux-x86_64
+
+        - os: windows-latest
+          os-arch: windows-x86_64
 
     env:
       FC: gfortran
-      GCC_V: 10
+      GCC_V: ${{ matrix.gcc_v }}
 
     steps:
     - name: Checkout code
       uses: actions/checkout@v1
 
+    - name: Install GFortran Linux
+      if: contains(matrix.os, 'ubuntu')
+      run: |
+        sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-${GCC_V} 100 \
+        --slave /usr/bin/gfortran gfortran /usr/bin/gfortran-${GCC_V} \
+        --slave /usr/bin/gcov gcov /usr/bin/gcov-${GCC_V}
+
     - name: Install fpm
       uses: fortran-lang/setup-fpm@v3
       with:
         github-token: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Fix fpm.toml (Ubuntu)
+      if: contains(matrix.os, 'ubuntu')
+      run: |
+        sed -i "s/^# link/link/" fpm.toml
 
     - name: Build
       run: |
@@ -26,5 +49,4 @@ jobs:
 
     - name: Search
       run: |
-        gfortran --version
         fpm run -- -V

--- a/fpm.toml
+++ b/fpm.toml
@@ -22,3 +22,6 @@ source-dir="src"
 name="fpm-search"
 source-dir="app"
 main="main.f90"
+
+[build]
+# link = "dl"

